### PR TITLE
New package: hamlib-3.3

### PR DESCRIPTION
**Pull request by bra1nwave**

### CODE REVIEW

**Comment by newbluemoon** [comment]
Python detection when cross-compiling can be fixed similar to brltty. Add

```
export PYTHON_CPPFLAGS="-I${XBPS_CROSS_BASE}/${py3_inc}"
export PYTHON_LIBS="-L${XBPS_CROSS_BASE}/usr/lib -lpython${py3_ver}${py3_abiver}"
```

to `pre_configure()` and change `python3-devel` to just `python3` in `hostmakedepends`.

However, there’s a problem with cross-building the perl bindings then.

```
make[3]: Entering directory '/builddir/hamlib-3.2/bindings'
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Hamlib.bs blib/arch/auto/Hamlib/Hamlib.bs 644
armv7l-linux-gnueabihf-gcc -c  -DIN_HAMLIB -I../include -I../src -I../lib -I../bindings -I/usr/armv7l-linux-gnueabihf//usr/include/python3.6m  -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe -g -Wall -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe -g   -DVERSION=\"3.2\" -DXS_VERSION=\"3.2\" -fPIC "-I/usr/lib/perl5/core_perl/CORE"   hamlibperl_wrap.c
cc1: error: unrecognized -mtune target: generic
```

MakeMaker uses the wrong flags, but this can be fixed in `bindings/Makefile.PL` by inserting Void’s flags, like e.g.:

```
sed -i -e "s;#'INC'.*;'CCFLAGS' => '${CFLAGS}',\n\t'OPTIMIZE' => '${CFLAGS}',;" bindings/Makefile.PL
```

So to not be more confusing here’s a diff of what I did to make it build locally ;)

```
diff --git a/srcpkgs/hamlib/template b/srcpkgs/hamlib/template
index 355681a519..8d852e4d34 100644
--- a/srcpkgs/hamlib/template
+++ b/srcpkgs/hamlib/template
@@ -5,7 +5,7 @@ revision=1
 build_style=gnu-configure
 configure_args="--disable-static --with-python-binding --with-perl-binding
  --with-tcl-binding --with-xml-support"
-hostmakedepends="pkg-config swig perl python3-devel automake libtool"
+hostmakedepends="pkg-config swig perl python3 automake libtool"
 makedepends="libusb-devel libxml2-devel python3-devel readline-devel tcl-devel"
 depends="perl"
 short_desc="Library to control radio transceivers and receivers"
@@ -16,6 +16,9 @@ distfiles="${SOURCEFORGE_SITE}/hamlib/${version}/${pkgname}-${version}.tar.gz"
 checksum=b55cb5e6a8e876cceb86590c594ea5a6eb5dff2e30fc13ce053b46baa6d7ad1d
 
 pre_configure() {
+       sed -i -e "s;#'INC'.*;'CCFLAGS' => '${CFLAGS}',\n\t'OPTIMIZE' => '${CFLAGS}',;" bindings/Makefile.PL
+       export PYTHON_CPPFLAGS="-I${XBPS_CROSS_BASE}/${py3_inc}"
+       export PYTHON_LIBS="-L${XBPS_CROSS_BASE}/usr/lib -lpython${py3_ver}${py3_abiver}"
        autoreconf -fi
 }
```

**Comment by bra1nwave** [comment]
Thanks man. I really appreciate it.
This is even better than the solution I wanted to do...

**Comment by newbluemoon** [comment]
Glad it worked out. :)

**Comment by newbluemoon** [comment]
Did you by any chance remove the python binding because of a configure error `Could not link test program to Python.`?  I didn’t get this erlier. It cannot link to libintl.so now. By adding gettext-devel to `makedepends` it works again.

**Comment by bra1nwave** [comment]
Yeah, I removed because I wanted to test something and forgot to add it back.
Thank you for your help!

File: srcpkgs/hamlib/template [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+depends="perl"
+short_desc="Library to control radio transceivers and receivers"
+maintainer="bra1nwave <brainwave@openmailbox.org>"
+license="LGPL-2.1-or-later, GPL-2.0-or-later"
+homepage="https://sourceforge.net/projects/hamlib/"
+distfiles="${SOURCEFORGE_SITE}/hamlib/${version}/${pkgname}-${version}.tar.gz"
+checksum=c90b53949c767f049733b442cd6e0a48648b55d99d4df5ef3f852d985f45e880
+
+pre_configure() {
+	CFLAGS+=" -DNO_POSIX_2008_LOCALE -D_GNU_SOURCE"
```
Comment (newbluemoon) [inline srcpkgs/hamlib/template]: `-DNO_POSIX_2008_LOCALE` is set in the perl template only for musl. Is it perhaps better to use it here also only conditionally?

**Comment by maxice8** [comment]
please fix conflicts

**Pull request closed without merging**